sqldelight-studio-plugin/src/main/kotlin/com/squareup/sqldelight/lang/SqlDelightFileViewProviderFactory.kt

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (C) 2016 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.sqldelight.lang
+
+import com.intellij.lang.Language
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.command.WriteCommandAction
+import com.intellij.openapi.vfs.LocalFileSystem
+import com.intellij.openapi.vfs.VirtualFile
+import com.intellij.psi.FileViewProviderFactory
+import com.intellij.psi.PsiDocumentManager
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiManager
+import com.intellij.psi.SingleRootFileViewProvider
+import com.squareup.sqldelight.SqliteCompiler
+import com.squareup.sqldelight.generating.TableGenerator
+
+class SqlDelightFileViewProviderFactory : FileViewProviderFactory {
+  override fun createFileViewProvider(virtualFile: VirtualFile, language: Language,
+      psiManager: PsiManager, eventSystemEnabled: Boolean) =
+      SqlDelightFileViewProvider(virtualFile, language, psiManager, eventSystemEnabled)
+}
+
+class SqlDelightFileViewProvider(virtualFile: VirtualFile, language: Language,
```

Comment:
(I don't really know if the public API of this module matters anyway. I'm going to stop caring unless you say otherwise.)